File: ios-app/ios-app/DNDDiceRoller/Models/DiceType.swift
import Foundation

enum DiceType: CaseIterable {
    case d4, d6, d8, d10, d12, d20, d100
    
    var sides: Int {
        switch self {
        case .d4: return 4
        case .d6: return 6
        case .d8: return 8
        case .d10: return 10
        case .d12: return 12
        case .d20: return 20
        case .d100: return 100
        }
    }
    
    var emoji: String {
        switch self {
        case .d4: return "🔸"
        case .d6: return "⚀"
        case .d8: return "🔶"
        case .d10: return "🔟"
        case .d12: return "🔷"
        case .d20: return "🎲"
        case .d100: return "💯"
        }
    }
    
    var displayName: String {
        return "d\(sides)"
    }
}

struct RollResult {
    let diceType: DiceType
    let result: Int
    let breakdown: String
    let timestamp: Date
    
    init(diceType: DiceType, result: Int, breakdown: String = "") {
        self.diceType = diceType
        self.result = result
        self.breakdown = breakdown
        self.timestamp = Date()
    }
}

File: android-app/android-app/app/src/main/kotlin/com/tiation/dnddiceroller/data/DiceType.kt
package com.tiation.dnddiceroller.data

enum class DiceType(val sides: Int, val emoji: String) {
    D4(4, "🔸"),
    D6(6, "⚀"),
    D8(8, "🔶"),
    D10(10, "🔟"),
    D12(12, "🔷"),
    D20(20, "🎲"),
    D100(100, "💯");
    
    override fun toString(): String = "d$sides"
}

data class RollResult(
    val diceType: DiceType,
    val result: Int,
    val breakdown: String = "",
    val timestamp: Long = System.currentTimeMillis()
)